hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/QueryConverter.java

```diff
@@ -156,40 +162,23 @@ private HazelcastSqlToRelConverter createSqlToRelConverter() {
      *  It is used instead of "expand" flag due to bugs in Calcite (see {@link #EXPAND}).
      * </li>
      * <li>
-     *  Transformation of distinct UNION to UNION ALL, merging the neighboring UNION relations.
-     * </li>
-     * <li>
      *  Check, if the relation uses cyclic user types, and if they are allowed - skip this step.
      * </li>
-     *
      * </ul>
      *
      * @param rel                       Initial relation.
      * @param cyclicUserTypesAreAllowed Whether cyclic user types are allowed.
      * @return Resulting relation.
+     * @implNote we're using Calcite's {@link HepProgramBuilder} to GC the rules we don't need earlier.
```

Comment:
I do not understand this note. I thought we keep the program in static variable so it is not GC'ed

Reply:
It is unrelated already, but previously I left the first phase rewrite in runtime.